.github/workflows/main.yml

```diff
@@ -0,0 +1,126 @@
+name: Singleton-Python-Client
+
+on: [pull_request, push]
+
+jobs:
+  pre-process:
+    name: Pre process
+    runs-on: ubuntu-latest
+    outputs:
+      were-only-docs-updated: ${{ steps.were-only-docs-updated-action.outputs.were-only-docs-updated }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2.
+      - name: Get commit range
+        id: get-commit-range-action
+        uses: ./.github/actions/get-commit-range-action
+      - name: Were only docs updated
+        id: were-only-docs-updated-action
+        uses: ./.github/actions/were-only-docs-updated-action
+        with:
+          commit-range: ${{ steps.get-commit-range-action.outputs.commit-range }}
+
+  check-header:
+    name: Check Header
+    runs-on: ubuntu-latest
+    needs: pre-process
+    if: needs.pre-process.outputs.were-only-docs-updated != 'yes'
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 2.
+      - name: Get commit range
+        id: get-commit-range-action
+        uses: ./.github/actions/get-commit-range-action
+      - name: Check header
+        run: |
+          git clone --branch=devops https://github.com/vmware/singleton.git devops
+          cp $GITHUB_WORKSPACE/devops/check_headers.py .
+          chmod +x check_headers.py
+          git diff ${{ steps.get-commit-range-action.outputs.commit-range }} --stat
+          git diff --name-only --diff-filter=d ${{ steps.get-commit-range-action.outputs.commit-range }}
+          python ./check_headers.py -f "$(git diff --name-only --diff-filter=d ${{ steps.get-commit-range-action.outputs.commit-range }})"
+
+  unit-test:
+    name: Unit Test
+    runs-on: ubuntu-latest
+    needs: pre-process
+    if: needs.pre-process.outputs.were-only-docs-updated != 'yes'
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Setup Python environment
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+      - name: Unit test
+        run: |
+          python -m pip install --upgrade pip
+          pip install coverage
+          pip install coverage pyyaml
+          cd sgtn4python/test
+          str=$(printf '=%.0s' {1..50})
+          echo $str Unit Test Start $str
+          coverage run --source='.',I18N,sgtn_bykey,sgtn_client,sgtn_debug,stgn_icu,sgtn_locale,sgtn_properties,sgtn_py_base,sgtn_py2_base,sgtn_py3_base,sgtn_py37_base,sgtn_util --omit='test_*.py','__init__.py' -m unittest
```

Comment:
If you set '.' as source which measn all modules, you don't need to list each module filename. Or if you want to specify the files to be calculated, you don't need this '.'